Topics/User_Experience/Usability_Heuristics.md

```diff
@@ -1,60 +1,236 @@
+
+"Mistakes are conscious errors, and often (though not exclusively) arise when a user has incomplete or incorrect information about the task, and develops a mental model that doesn’t match how the interface actually works." (9)
+
+__How can we implement this?__
+Reducing Slips:
+
+As stated in the article, slip-type mistakes often happen when the user knows how to achieve their goal but make mistakes when trying to achieve it. While the user is in control, a successful software application may reduce slips by providing guidance and correction when slips occur and allowing the user to rectify problems.
 
-**Example:** When a user clicks exit on an application, a prompt asking "Are you sure you want to exit?" will pop up.
+- User Constraints: Constraining the user in some circumstances can reduce the likelihood of error. For example, when the user is asked for their date of birth the system should reject the answer if that date is in the future; we may further implement input-validation features to ensure that nonsensical information (like incorrect postal codes) is not entered and that formatting is correct.
+
+- Offer Suggestions: We may provide suggestions to the user based on context. A website may offer completed results in the search bar based on what the user has entered instead of requiring an exact match.
+
+- Choose Good Defaults: Based on established user stories, we can offer defaults that fit an expected user requirement. The user will not need to specify all information if they use this default, eliminating the potential for error. For example, a website for booking restaurant reservations may suggest available timeslots.
+
+- Use Forgiving Formatting: "Some tasks really do require users to type very detailed or precise information, but forcing people to provide this information in a very specific format can be at odds with good usability practices: If you are asking users to input numerical information into a form, be flexible, and format that information in a way that is easily scannable (by humans, not machines) in order to prevent mistakes." (9) We may format user input automatically, or sanitize user input, or provide clear restrictions.
+
+Reducing Mistakes:
+
+When users interact with an application, they have a goal they want to achieve. Based on the user's understanding of how the system operates, the user forms a plan. Then they take action based on this plan, and see whether their goal has been achieved. Mistakes occur when there is a mismatch between user understanding of the software and how the software actually works, leading the user to create a plan that results in an unexpected outcome (10). Generally, to reduce mistakes it is necessary to both convey the designer's model of the application to the user and to design the application in such a way that it is intuitive and easy to learn.
+
+
+**Example:** 
```

Comment:
You gave some nice examples earlier so I don't think you need to include this line.

Reply:
(The line I'm referring to is line 129 if it is unclear)

Reply:
Thank you. I will remove this example, as it is not contributing anything novel to the document.